open-craft/opencraft
# BB-1848: Fix styling consistency and expired token issue.

This PR fixes a few bugs and inconsistencies in theming in one of multiple pages and also a bug which caused a infinite loop.

**Testing instructions:**
bfaa87a:
1. Log in the console.
2. Open Chrome tools, edit the access and refresh token (put invalid values, not empty ones).
3. Check that there's no infinite loop and you're redirected to the login page.

6936715:
1. Log in the new console, check that you are redirected to the theme preview and colors page.
2. Click through all pages in the side menu, and check that there's no weird resizing or changes in scale happening.
3. Check that the header image covers the entire header.
![heaer](https://user-images.githubusercontent.com/27893385/82083202-b3f02080-96bf-11ea-8452-c3be39a1d7f9.png) 

3b027a9:
1. Check that the footer preview component is correctly displayed.

d909aeb:
1. Log in the console.
2. Hijack the backend code to test the multiple responses with the values below. Code here: 
https://github.com/open-craft/opencraft/blob/86091ae992a8f4b214ab5f0cc8796ed3f6495f20/registration/api/v2/views.py#L539.
3. Set and check if the status show as up-to-date and deploy button is disabled.
```
       data = {
            'undeployed_changes': [],
            'deployed_changes': deployed_changes,
            'status': DeploymentState.healthy.name,
            'deployment_type': deployment_type,
        }
```
4. Set and check if the status show as up-to-date and deploy button is enabled (make sure you have some changes pending).
```
       data = {
            'undeployed_changes': undeployed_changes,
            'deployed_changes': deployed_changes,
            'status': DeploymentState.healthy.name,
            'deployment_type': deployment_type,
        }
```
5. Set and check if the cancel redeployment button is enabled.
```
       data = {
            'undeployed_changes': undeployed_changes,
            'deployed_changes': deployed_changes,
            'status': DeploymentState.provisioning.name,
            'deployment_type': DeploymentType.user,
        }
```
6. Set and check if the cancel redeployment button is disabled.
```
       data = {
            'undeployed_changes': undeployed_changes,
            'deployed_changes': deployed_changes,
            'status': DeploymentState.provisioning.name,
            'deployment_type': DeploymentType.admin,
        }
```

**Reviewers:**
- [ ] Anyone at OpenCraft
